src/transformers/modeling_utils.py

```diff
@@ -495,6 +495,18 @@ def tie_weights(self):
                 self = getattr(self, self.base_model_prefix)
             self._tie_encoder_decoder_weights(self.encoder, self.decoder, self.base_model_prefix)
 
+    @staticmethod
+    def _tie_encoder_decoder_word_embeddings(encoder: nn.Module, decoder: nn.Module):
+        encoder_embeddings = encoder.get_input_embeddings()
+        decoder_embeddings = decoder.get_input_embeddings()
+        if encoder_embeddings is not None and decoder_embeddings is not None:
+            if decoder_embeddings.weight.size() == encoder_embeddings.weight.size():
```

Comment:
I wonder if it's better to throw a warning if this clause is `False`?

Reply:
I think it is better to notify when it is `True`, because when user inits encoder-decoder from bert and gpt, he doesn't expect embeddings to be tied and they aren't, but in bert2bert case it is better to know that they are.

Reply:
I think there are two nice features we could add here:

1) If `decoder_embeddings.weight.size() != encoder_embeddings.weight.size()`, then let's set `self.config.tie_encoder_decoder_word_embeds = False` so that it is saved as `False` automatically afterwards. For this I would also advocate for adding a `logger.warn(...)` or `logger.info(...)` with something like `"Word embeddings of encoder and decoder have a different sizes. Setting `config.tie_encoder_decoder_word_embeds` to `False`..."`

2) I think we should also add a `logger.info(...)` if "`type(encoder.config.__class__) != type(decoder.config.__class__)` saying that  encoder and decoder are different classes and that the user should make sure that tying the word embedding matrices is the desired behavior and if not he should disable it with `model.config.tie_encoder_decoder_word_embeds = False`. This will be shown everytime, but I think it's worth / necessary because there will be cases were two models that should not have their word embeddings tied will have tied word embeddings because by chance they have the same word embeddings.